packages/@aws-cdk/cdk/lib/aspects/aspect.ts

```diff
@@ -0,0 +1,52 @@
+import { IConstruct } from '../core/construct';
+
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * The type of Aspect
+   */
+  readonly type: string;
+
+  /**
+   * All aspects can visit by IConstructs
+   */
+  visitTree(node: IConstruct): void;
```

Comment:
@eladb let's take a good look here to make sure I implemented what you want. I *think* I want a method that calls the action an Aspect author creates. I want this method so I can manage the checking of the visit status. If that checking was in `visit()` then the author would have to remember to implement it. (this is implemented in the PR currently)

The other option is that we could move the management of which aspect were visited by which node, to the node itself. This is going to add more to ConstructNode but then it's clearly hooked on to the framework.

Reply:
+1 for moving to the framework

Reply:
Ok, let me make that move. I don't think it will be very complicated.